#164688417 user/admin can delete answer

Amoswachira opened the pull request:
#### What does this PR do?
-Delete  answers

## What are the tasks to be completed?
- Ensure the following endpoints is working:-
```
1. DELETE /api/meetups/{meetupId}/questions/{questionId}/answers/{answerId}/
```

## How can this be manually tested?
### Local Development Setup
- First Create python virtual env
```
 $ virtualenv -p python .venv
```
- Install Requirements
```
 $ pip install -r requiremts.txt
```
- Copy .env-example to .env and set config
```
 $ cp .env-example .env
```
- Create postgres database
```
 $ sudo su postgres
 $ psql
 postgres=# CREATE USER django WITH PASSWORD 'password';
 postgres=# ALTER ROLE django SET client_encoding TO 'utf8';
 postgres=# ALTER ROLE django SET default_transaction_isolation TO 'read committed';
 postgres=# ALTER ROLE django SET timezone TO 'UTC';
 postgres=# CREATE DATABASE drf;
 postgres=# GRANT ALL PRIVILEGES ON DATABASE drf TO django;
 postgres=# \q
 $ exit
```
- Create admin user
    1. cd api/
    2. python manage.py createsuperuser
- Create User using your email and password

- Run Server
```
$ python manage.py runserver
```
a. After cloning the repo:-
```
git checkout ft-delete-answer-164688417
```
b. Create meetups

- Follow the instructions [here](https://github.com/Ogutu-Brian/questioners/pull/6) to create meetups

c. Create a Question
- Follow the instructions [here](https://github.com/Ogutu-Brian/questioners/pull/14) to create a question

d. Create an Answer
- Follow the instructions [here](https://github.com/Ogutu-Brian/questioners/pull/23) to Create an Answer
e. Delete an Answer

- Use swagger for testing
#### What are the relevant pivotal tracker stories?
#164688417
#### Screenshots (if appropriate)
![delete answer](https://user-images.githubusercontent.com/17140636/54722941-f333e780-4b76-11e9-9f76-6c26df1fdd40.JPG)

IssaIan commented on api/answers/views.py:
```diff
@@ -129,3 +129,43 @@ def put(self, request, meetupId, questionId, answerId):
+        try:
+            answer = Answer.objects.get(id=answerId)
+            userid = request.user.id
+            if userid != answer.creator.id:
```
Ensure to also validate that if a user is an admin,  the user can delete any answer posted.

Amoswachira replied:
@IssaIan  i fixed it

IssaIan reviewed (changes requested):
Ensure to validate that an admin user can delete any answer too.

Bakley commented:
Just solve merge conflicts @Amoswachira

Bakley reviewed (approved):
Good job @Amoswachira

Ogutu-Brian reviewed (approved):
Good work,
Just implement the feedback from @IssaIan  and resolve the merge conflicts

Ogutu-Brian commented on api/config/swagger.py:
```diff
@@ -487,8 +498,30 @@ def schema_view_swagger(request):
                     ],
                     description = 'Update an existing answer'
                 ),
+                'Delete answer': coreapi.Link(
```
Rebase and test the documentation,
If this is added to the line specified in develop it will be under answers instead of questions

Amoswachira replied:
@Ogutu-Brian I have fixed it

Amoswachira committed:
ft(answer):delete answer

-rebase and implement feedback

-implement pr feedback

-delete answer

-delete a specific answer

-fix migrations

-fix delete not allowed

-add tests

-fix meetup migrations

-fix the final migration error

[Delivers #164688417]

Amoswachira commented:
@IssaIan  @Ogutu-Brian  i have fixed everything

Bakley reviewed (commented):
@Amoswachira Good job on the tests, I checked it out and liked it. I also tested the application using postman and it works as specified.
